src/main.c

```diff
@@ -545,13 +549,15 @@ int main(int argc, char **argv)
 
 	// https://wiki.ubuntu.com/SecurityTeam/Specifications/SnappyConfinement
 
+#ifdef STRICT_CONFINEMENT
 	int rc = 0;
 	// set apparmor rules
 	rc = aa_change_onexec(aa_profile);
 	if (rc != 0) {
 		if (secure_getenv("SNAPPY_LAUNCHER_INSIDE_TESTS") == NULL)
 			die("aa_change_onexec failed with %i", rc);
 	}
+#endif
 	// set seccomp (note: seccomp_load_filters die()s on all failures)
 	seccomp_load_filters(aa_profile);
```

Comment:
Why aren't the other sandboxing features disabled as well? For example, seccomp needs a particular kernel and very soon will need a newer kernel/particular patches backported to work with snapd designs. '--disable-confinement' suggests to me disabling much more than apparmor. It should disable apparmor, seccomp, devpts, udev/cgroups and private mount. I would imagine that with --disable-confinement you would want to also strip the setuid bit as well. I'm not sure how useful it would be for people bootstrapping the confinement story, but perhaps in a future commit you could allow things like:

- --disable-confinement-apparmor # disables apparmor
- --disable-confinement-cgroups # disables cgroups
- --disable-confinement-devpts # disables devpts
- --disable-confinement-private-tmp # disables private tmp
- --disable-confinement-seccomp # disables seccomp
- --disable-confinement # disables everything in one go

Please don't block on --disable-confinement-* though.